tests/unit/services/sl-modal-test.js

```diff
@@ -0,0 +1,66 @@
+import Ember from 'ember';
+import { moduleFor, test } from 'ember-qunit';
+import ModalService from 'sl-ember-components/services/sl-modal';
+
+let service = ModalService.create();
+
+let MockModal = Ember.Object.extend({
+    name: 'mock',
+    isOpen: true
+});
+
+moduleFor( 'service:sl-modal', 'Unit | Service | sl modal', {
+    unit: true,
+
+    beforeEach() {
+        //reset modals property before each test so tests are independent
+        service.set( 'modals', {} );
+    }
+});
+
+test( 'Register a modal and find it', function( assert ) {
+    let mockModal = MockModal.create();
+
+    service.register( mockModal, mockModal.name );
+
+    let modal = service.find( mockModal.name );
+
+    assert.equal( modal.name, mockModal.name );
+});
+
+test( 'Registering the same modal twice throws an error', function( assert ) {
```

Comment:
If register() fails and does not throw an error then this test will fail not with a failed assertion but with an `Expected at least one assertion, but none were run - call expect(0) to accept zero assertions.` error.  To prevent this approach this testing instead like https://github.com/softlayer/sl-ember-components/blob/develop/tests/unit/components/sl-tooltip-test.js#L42-L53 so that an assertion always exists, and not just in the catch block of a try/catch.